pkg/controller/cheinstallation/cheinstallation_controller.go

```diff
@@ -215,6 +220,10 @@ func (r *ReconcileCheInstallation) ensureCheSubscription(logger logr.Logger, ns
 }
 
 func (r *ReconcileCheInstallation) ensureCheCluster(logger logr.Logger, ns string, cheInstallation *v1alpha1.CheInstallation) (bool, string, error) {
+	if err := r.addCheCluserWatch(logger); err != nil {
```

Comment:
wouldn't it be cleaner to watch the CRDs and call this func when the `checlusters.org.eclipse.che` CRD is created?

Reply:
yes, I agree - with this it would be much easier and safer the catch the correct moment when we should start watching the resource - without any errors etc...

Reply:
See the discussion at https://coreos.slack.com/archives/CHK0J6HT6/p1575605828426700?thread_ts=1575537980.370000&cid=CHK0J6HT6